docs/webapp.adoc

```diff
@@ -1,30 +1,625 @@
+
+{zwsp} +
+
+Keep in mind that except for the XP services in the second variation, *no running XP is strictly necessary for this to work*. The data-serving endpoint could be any API (e.g. REST) instead of guillotine, and the initial HTML and JS/CSS assets are static and could be served from anywhere. Use whatever approach suits your project.
+
+But we'll use XP anyway in this chapter: we already have it up and running from the previous chapters. So we'll use link:https://developer.enonic.com/guides/my-first-webapp[the XP webapp functionality] (see link:https://developer.enonic.com/docs/xp/stable/runtime/engines/webapp-engine[here] for more documentation) to serve the initial HTML, and the link:https://developer.enonic.com/docs/xp/stable/runtime/engines/asset-service[regular XP asset functionality] for serving the assets for the entries and dependencies.
+
+{zwsp} +
 
 === Source files
 
-.Files involved (src/main/resources/...):
+.Files involved (src/main/resources/...) - in addition to <<guillotine#chapter_source_files, the ones from chapter 8>>:
 [source,files]
 ----
+webapp/
+    webapp.es6
+    webapp.html
+
+assets/webapp/
+    script.es6
 ----
-<1> We're going to
+
+XP uses _webapp.es6_ and _webapp.html_ to generate an initial HTML that directly makes the browser run most of <<#nutshell, the steps above>>, fetching assets and setting up initial values, and then calling the final _script.es6_ asset, which handles the rest.
 
 {zwsp} +
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+{zwsp} +
+{zwsp} +
 {zwsp} +
+
+
+== 1. Completely standalone
+
+In this first of the two approaches, we'll minimize the use of runtime XP: all values and asset URLs are hardcoded. XP is used to serve the initial HTML and the assets, as well as providing a guillotine endpoint where the browser can fetch data, but this functionality can easily be replaced with any other file- and data-serving solution and still work just fine.
+
+[[html1]]
+=== HTML base
+
+The webapp begins with some basic HTML, setting it all up in the browser.
+
+.webapp.html:
+[source,html,options="nowrap"]
+----
+<!DOCTYPE html>
+<html>
+    <head>
+        <meta charset="UTF-8">
+        <title>Completely standalone</title>
+
+
+        <!--1-->
+        <script crossorigin src="https://unpkg.com/react@16/umd/react.production.min.js"></script>
+        <script crossorigin src="https://unpkg.com/react-dom@16/umd/react-dom.production.min.js"></script>
+
+        <!--2-->
+        <script src="(...) /react4xp/shared.d366a6c1b.js"></script>
+        <link rel="stylesheet" type="text/css" href="(...) /react4xp/shared.5440dda80.css" />
```

Comment:
Is it possible to avoid (...) ? People will copy paste from the document.
Maybe just a {REPLACE ME} ?

Reply:
Ah, this was a forgotten left-behind, good catch. 

I wrote a similar placeholder thing in previous chapters, but then as  `(...my.app.service)`. Here it should have been something like `(...my.app.assets)`, does that sound okay?

Reply:
Using `(...my.asset.url)` in the links and updated the note:

"As before, the asset path URLs depend on your setup - and this applies to both the *content-dependent {hash}* in the filename and the *asset path* itself - whatever the root path to your asset source is. For example, on my local machine running XP, the `(...my.asset.url)` part of the path looks like this:

`/_/asset/com.enonic.app.react4xp:1604314030`

...and the part `:1604314030` can even be skipped (at the expense of caching). If you've run through the previous chapter, you can for example copy the corresponding asset-URLs from the produced page source HTML in the preview."

Reply:
✅ ❓

Reply:
Look good. The important part is that people don't build and get stuck.